### PR TITLE
sampling bug fix in diffusers tutorial "basic_training.md"

### PR DESCRIPTION
**Pull request by yue-here**
# What does this PR do?

In the diffusers basic training tutorial, setting the manual seed argument (generator=torch.manual_seed(config.seed)) in the pipeline call inside the evaluate() function rewinds the random state of the dataloader shuffling, leading to overfitting due to the model seeing same sequence of training examples after every evaluation call.

Using generator=torch.Generator(device='cpu').manual_seed(config.seed) avoids this.

Fixes # [7991](https://github.com/huggingface/diffusers/issues/7991)

- Docs: @stevhliu and @sayakpaul

### CODE REVIEW

**Review by stevhliu — APPROVED** [review]
Thanks for the fix! Once we fix the failing test, I'll merge your PR 🙂
